src/test/ui/closures/2229_closure_analysis/migrations/precise_no_migrations.rs

```diff
@@ -0,0 +1,105 @@
+// run-pass
+
+#![deny(disjoint_capture_drop_reorder)]
+
+#[derive(Debug)]
+struct Foo(i32);
+impl Drop for Foo {
+    fn drop(&mut self) {
+        println!("{:?} dropped", self.0);
+    }
+}
+
+struct ConstainsDropField(Foo, Foo);
+
+// Test that if all paths starting at root variable that implement Drop are captured
+// then it doesn't trigger the lint.
```

Comment:
It occurs to me that the *order* in which these drops will execute *may* be different -- but I think that's ok. It seems pretty unlikely that this will be relevant. Still, I think we should document these assumptions -- maybe in the project-rfc-2229 repo? We're going to want to note them down in the Edition Migration Guide.

Reply:
(i.e., before, the closure would drop `t`, which would in turn drop its fields in order, but now the order of the fields in the closure may not match -- *actually*, if we wanted, we could sort the fields in the closure and fix that, couldn't we?)

Reply:
Yea sorting won't be hard, 

For place_a, place_b, let `c` length of the common starting prefix of these places. Eg: for p.x.y and p.x.z then c = 2.

Then we can write a compartor which is essentially `place_a.projections[c] < place_b.projections[c] =>  place_a < place_b`